File: scheduler/src/main/java/ch/fhnw/edu/eaf/scheduler/domain/EventAttendee.java
package ch.fhnw.edu.eaf.scheduler.domain;

import javax.validation.constraints.NotNull;
import java.io.Serializable;

/**
 * An attendee of an event.
 */
public class EventAttendee {

    /**
     * The different sandwich-types available.
     */
    public enum FoodType {
        VEGI,
        NORMAL,
        NONE
    }

    /**
     * The id of the event-attendee.
     */
    public Long Id;

    /**
     * The event the current attendee is attending.
     */
    public Event event;

    /**
     * The user that attends the event.
     */
    public User user;

    /**
     * The foodtype chosen by the user/attendee.
     */
    public FoodType foodType;

    /**
     * Indicates whether or not an attendee wants a drink or not.
     */
    public boolean drink;
}

File: ionic/src/models/EventAttendee.ts
import {FoodType} from "./FoodType";

export class EventAttendee {
  /**
   * EventAttendee id
   */
  id: number;

  /**
   * The resource_uri of the user that is attending
   */
  user: string;

  /**
   * The resource_uri of the event the attendee is attending
   */
  event: string;

  /**
   * What kind of food did the attendee order?
   */
  foodType: FoodType;

  /**
   * Did the attendee order a drink?
   */
  drink: boolean;
}
